workspaces/ui-v2/src/optic-components/pages/diffs/ReviewEndpointDiffPage/ReviewEndpointDiffPage.tsx

```diff
@@ -70,30 +73,56 @@ export const ReviewEndpointDiffPage: FC<ReviewEndpointDiffPageProps> = ({
     sideEffect: (newName: string) => debouncedSetName(endpointId, newName),
   });
   const endpointId = getEndpointId({ method, pathId });
+  const getNextIncompleteDiff = (): number =>
+    Math.max(
+      0,
+      // findIndex possibly returns -1 if all diffs are handled,
+      // in which case useRedirectForDiffCompleted should redirect back
+      shapeDiffs.findIndex(
+        (shapeDiff) =>
+          !(
+            shapeDiff.diffDescription?.diffHash &&
+            isDiffHandled(shapeDiff.diffDescription.diffHash)
+          )
+      )
+    );
 
-  const allDiffsHandled = shapeDiffs.every(
-    (shapeDiff) =>
-      shapeDiff?.diffDescription?.diffHash &&
-      isDiffHandled(shapeDiff.diffDescription.diffHash)
-  );
-  const [currentIndex, setCurrentIndex] = useState(0);
+  const [currentIndex, setCurrentIndex] = useState(getNextIncompleteDiff());
   const [previewCommands, setPreviewCommands] = useState<any[]>([]);
+  const renderedDiff = shapeDiffs[currentIndex];
 
   return (
     <TwoColumnFullWidth
       left={
-        allDiffsHandled ? (
```

Comment:
So I realized that this view was never rendered, given that we _always_ redirect when all the diffs are handled